stock_picking_accounting_date/models/stock_move.py

```diff
@@ -46,3 +46,27 @@ def _prepare_account_move_vals(
         if self.accounting_date:
             am_vals.update({"date": self.accounting_date})
         return am_vals
+
+    def _get_price_unit(self):
+        """Set date for convert price unit multi currency."""
+        self.ensure_one()
+        price_unit = super()._get_price_unit()
+        if (
+            hasattr(self, "purchase_line_id")
+            and self.accounting_date
+            and not self.origin_returned_move_id
+            and self.purchase_line_id
+            and self.product_id.id == self.purchase_line_id.product_id.id
+        ):
+            line = self.purchase_line_id
+            order = line.order_id
+            price_unit = line.price_unit
+            if order.currency_id != order.company_id.currency_id:
+                price_unit = order.currency_id._convert(
+                    price_unit,
+                    order.company_id.currency_id,
+                    order.company_id,
+                    self.accounting_date,
+                    round=False,
+                )
+        return price_unit
```

Comment:
As we just talked, let's do something like this, and extend `_convert()` of the currency to use the date from the context if available.
```suggestion
    def _get_price_unit(self):
        self.ensure_one()
        if self.accounting_date:
            self = self.with_context(accounting_date=self.accounting_date)
        return super()._get_price_unit()
```

Reply:
I believe we don't need to add `if self.accounting_date:` because `accounting_date` will always have value.